src/bitset.rs

```diff
@@ -20,10 +24,10 @@ pub const SHIFT3: usize = SHIFT2 + BITS;
 /// Adding beyond this limit will cause the `BitSet` to panic.
 #[derive(Clone)]
 pub struct BitSet {
-    layer3: u32,
-    layer2: Vec<u32>,
-    layer1: Vec<u32>,
-    layer0: Vec<u32>,
+    layer3: usize,
```

Comment:
It hurt a bit unfortunately, it was minor but measurable. `ecs_bench` went from 60->65us, no change on the build times.